### PR TITLE
Match 4 functions in BGcheck, remove nonequivalency in DynaPoly_ExpandSRT and BgHakaBombwall

### PR DESCRIPTION
**Pull request by petrie911**
Still some nasty stuff left, of course.

### CODE REVIEW

File: src/code/z_bgcheck.c [hunk 1]
```diff
@@ -4425,27 +4427,24 @@ s32 WaterBox_GetSurfaceImpl(PlayState* play, CollisionContext* colCtx, f32 x, f3
...
-                            *outWaterBox = curWaterBox;
-                            *ySurface = curWaterBox->minPos.y;
-                            *bgId = i;
-                            return true;
-                        }
+        if (!(colCtx->dyna.bgActorFlags[i] & 1) || (colCtx->dyna.bgActorFlags[i] & 4)) {
+            continue;
+        }
+        bgActor = &colCtx->dyna.bgActors[i];
+        if (bgActor->colHeader->numWaterBoxes != 0 && bgActor->colHeader->waterBoxes != NULL) {
```
Comment (jpburnett) [inline src/code/z_bgcheck.c:4434]:
```suggestion
        if ((bgActor->colHeader->numWaterBoxes != 0) && (bgActor->colHeader->waterBoxes != NULL)) {
```

File: src/code/z_bgcheck.c [hunk 2]
```diff
@@ -4425,27 +4427,24 @@ s32 WaterBox_GetSurfaceImpl(PlayState* play, CollisionContext* colCtx, f32 x, f3
...
+            for (curWaterBox = colCtx->dyna.waterBoxList.boxes + bgActor->waterboxesStartIndex;
+                 curWaterBox <
+                 colCtx->dyna.waterBoxList.boxes + bgActor->waterboxesStartIndex + bgActor->colHeader->numWaterBoxes;
+                 curWaterBox++) {
+                if (curWaterBox->properties & 0x80000) {
+                    continue;
+                }
+                if (curWaterBox->minPos.x < x && x < curWaterBox->minPos.x + curWaterBox->xLength) {
```
Comment (jpburnett) [inline src/code/z_bgcheck.c:4442]:
```suggestion
                if ((curWaterBox->minPos.x < x) && (x < curWaterBox->minPos.x + curWaterBox->xLength)) {
```
Is that the intended way? I don't know if parenthesis can help separate the args to make it clearer

File: src/code/z_bgcheck.c [hunk 3]
```diff
@@ -4425,27 +4427,24 @@ s32 WaterBox_GetSurfaceImpl(PlayState* play, CollisionContext* colCtx, f32 x, f3
...
+                    if (curWaterBox->minPos.z < z && z < curWaterBox->minPos.z + curWaterBox->zLength) {
```
Comment (jpburnett) [inline src/code/z_bgcheck.c:4443]: same here

**Review by jpburnett — COMMENTED** [review]
Just some comments on adding some parenthesis to clear up the comparisons

File: src/code/z_bgcheck.c [hunk 4]
```diff
@@ -379,31 +380,41 @@ s32 CollisionPoly_CheckZIntersectApprox(CollisionPoly* poly, Vec3s* vtxList, f32
                                             y, zIntersect);
 }
 
-#ifdef NON_MATCHING
 s32 CollisionPoly_LineVsPoly(BgLineVsPolyTest* a0) {
     static Vec3f polyVerts[3]; // D_801EDB70
     static Plane plane;        // D_801EDB98
```
Comment (AngheloAlf) [inline src/code/z_bgcheck.c:385]: We don't really have a good way to deal with in-function static bss reordering, which will be a problem in the future, specially with the big amount of bss this file has.
My recommendation is to leave this function as NON_MATCHING with a note saying this function matches but bss

Comment (petrie911) [reply src/code/z_bgcheck.c:385]: Should the same be done with `SphVsPoly`, which already has its static bss in-functioned?

File: src/code/z_bgcheck.c (hunk 2 above)
Comment (petrie911) [reply src/code/z_bgcheck.c:4442]: It's the way I intended it. I thought I caught them all but apparently not.

File: src/code/z_bgcheck.c (hunk 4 above)
Comment (hensldm) [reply src/code/z_bgcheck.c:385]: > Should the same be done with `SphVsPoly`, which already has its static bss in-functioned?

I think so. I'm kind of surprised it hasn't been troublesome yet.
Note I've usually left the comment `// needs in-function static bss` for functions that match but just need in function bss. That way it is easier to find them later.

**Review by EllipticEllipsis — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]
The bss still worries me, but hopefully it will be fine

File: src/code/z_bgcheck.c (hunk 1 above)
Comment (hensldm) [reply src/code/z_bgcheck.c:4457]: Bump missing, parens.

File: src/code/z_bgcheck.c (hunk 4 above)
Comment (hensldm) [reply src/code/z_bgcheck.c:394]: Bump on engineer's comment, for these named bss variables, can you prefix `s`

**Review by hensldm — COMMENTED** [review]
Bump on other's comments

**Review by hensldm — APPROVED** [review]

**Pull request merged**